streams/src/main/java/org/apache/kafka/streams/state/internals/CachingKeyValueStore.java

```diff
@@ -128,9 +134,8 @@ public boolean isOpen() {
     @Override
     public synchronized V get(final K key) {
         validateStoreOpen();
-        if (key == null) {
-            return null;
-        }
+        validateKeyNotNull(key);
```

Comment:
cc @mjsax @guozhangwang if you have any other opinion. Streams shouldn't be getting any `null` keys so I'd prefer flagging this as an exception. E.g., if someone does a put(null, value), that should fail (in the streams context).